code/ai/aigoals.cpp

```diff
@@ -2349,14 +2433,32 @@ void ai_process_mission_orders( int objnum, ai_info *aip )
 	}	
 
 
+	// save the current goal (if any) first, in case it's wiped out by the next action
+	int current_goal_ai_mode = current_goal->ai_mode;
+	auto current_goal_target_name = current_goal->target_name;
+	auto current_goal_target_ship = current_goal_target_name ? ship_registry_get(current_goal_target_name) : nullptr;
+
+	if (current_goal->flags[AI::Goal_Flags::Clear_all_goals_first]) {
+		// stay-still, form-on-wing, and play-dead all clear their goals here...
+		// 
+		// clear out the object's goals.  Seems to me that if a ship is staying still for a purpose
+		// then we need to clear everything out since there is not a real way to get rid of this goal
+		// 
+		// for form on wing, we need to clear out all goals for this ship, and then call the form on wing AI code
+		// 
+		// if a ship is playing dead, MWA says that it shouldn't try to do anything else.
+		// 
+		// clearing out goals is okay here since we are now what mode to set this AI object to.
+
+		ai_clear_ship_goals(aip);
```

Comment:
Since `current_goal` is a dangling pointer after the call to `ai_clear_ship_goals()`, let's set it to `nullptr` here.